src/troute-network/troute/nhd_network_utilities_v02.py

```diff
@@ -798,3 +800,78 @@ def build_data_assimilation_folder(data_assimilation_parameters, run_parameters)
         usgs_df = pd.DataFrame()
 
     return usgs_df
+
+
+def build_refac_connections(diff_network_parameters):
+    '''
+    Construct network connections network for refacored dataset. 
+    
+    Arguments
+    ---------
+    diff_network_parameters (dict): User input network parameters
+    
+    Returns:
+    --------
+    connections (dict int: [int]): Network connections
+    '''
+
+    # crosswalking dictionary between variables names in input dataset and 
+    # variable names recognized by troute.routing module.
+    cols = diff_network_parameters.get(
+        'columns', 
+        {
+        'key'       : 'link',
+        'downstream': 'to',
+        'dx'        : 'Length',
+        'n'         : 'n',
+        'waterbody' : 'NHDWaterbodyComID',
+        'gages'     : 'gages',
+        'alt'      : 'z',
+        'line_d'    : 'xid_d',        
+        }
+    )
+
+    # read parameter dataframe 
+    param_df = nhd_io.read(pathlib.Path(diff_network_parameters["geo_file_path"]))
+
+    # numeric code used to indicate network terminal segments
+    terminal_code = set(param_df.to.unique()) - set(param_df.link.unique())
+
+    # select the column names specified in the values in the cols dict variable
+    param_df = param_df[list(cols.values())]
+
+    # rename dataframe columns to keys in the cols dict variable
+    param_df = param_df.rename(columns=nhd_network.reverse_dict(cols))
+
+    # set parameter dataframe index as segment id number, sort
+    param_df = param_df.set_index("key").sort_index()
+
+    # get and apply domain mask
+    if "mask_file_path" in diff_network_parameters:
```

Comment:
I believe I dropped this in my branch as it was causing issues when actually applied. Let's revisit after testing and we can use the code from my branch.